src/main/java/org/apache/sling/feature/cpconverter/handlers/BundleEntryHandler.java

```diff
@@ -109,111 +154,324 @@ public void handle(@NotNull String path, @NotNull Archive archive, @NotNull Entr
+    }
 
-                Objects.requireNonNull(converter.getArtifactsDeployer()).deploy(new InputStreamArtifactWriter(input), id);
+    @Nullable InputStream extractSlingInitialContent(@NotNull ArtifactId bundleArtifactId, @NotNull JarFile jarFile, @NotNull ContentPackage2FeatureModelConverter converter, @Nullable String runMode) throws Exception {
+        if (slingInitialContentPolicy == SlingInitialContentPolicy.KEEP) {
+            return null;
+        }
+        // parse "Sling-Initial-Content" header
+        Manifest manifest = Objects.requireNonNull(jarFile.getManifest());
+        Iterator<PathEntry> pathEntries = PathEntry.getContentPaths(manifest, -1);
+        if (pathEntries == null) {
+            return null;
+        }
+        logger.info("Extracting Sling-Initial-Content from '{}'", bundleArtifactId);
+        Collection<PathEntry> pathEntryList = new ArrayList<>();
+        pathEntries.forEachRemaining(pathEntryList::add);
+
+        // remove header
+        manifest.getMainAttributes().remove(new Attributes.Name(PathEntry.CONTENT_HEADER));
+        // change version to have suffix
+        Version originalVersion = new Version(Objects.requireNonNull(manifest.getMainAttributes().getValue(Constants.BUNDLE_VERSION)));
+        manifest.getMainAttributes().putValue(Constants.BUNDLE_VERSION, getModifiedOsgiVersion(originalVersion).toString());
+        Path newBundleFile = Files.createTempFile(converter.getTempDirectory().toPath(), "newBundle", ".jar");
+        
+        // create JAR file to prevent extracting it twice and for random access
+        JcrNamespaceRegistry namespaceRegistry = createNamespaceRegistry(manifest, jarFile, converter.getFeaturesManager().getNamespaceUriByPrefix());
+        
+        Map<PackageType, VaultPackageAssembler> packageAssemblers = new EnumMap<>(PackageType.class);
+        try (OutputStream fileOutput = Files.newOutputStream(newBundleFile, StandardOpenOption.WRITE, StandardOpenOption.TRUNCATE_EXISTING);
+            JarOutputStream bundleOutput = new JarOutputStream(fileOutput, manifest)) {
+            
+            for (Enumeration<JarEntry> e = jarFile.entries(); e.hasMoreElements();) {
+                JarEntry jarEntry = e.nextElement();
+                if (!jarEntry.isDirectory()) {
+                    try (InputStream input = jarFile.getInputStream(jarEntry)) {
+                        if (!extractSlingInitialContent(jarEntry, input, bundleArtifactId, pathEntryList, packageAssemblers, namespaceRegistry, converter)) {
+                            // skip manifest, as already written in the constructor (as first entry)
+                            if (jarEntry.getName().equals(JarFile.MANIFEST_NAME)) {
+                                continue;
+                            }
+                            // copy entry as is to the stripped bundle
+                            ZipEntry ze = new ZipEntry(jarEntry.getName());
```

Comment:
I'm not sure about this one - a JarEntry can have attributes (like e.g. last modified etc) which this code will remove. I think it would be better to use the original entry instead of a copy (and keep it a real JarEntry).

Reply:
Done in https://github.com/apache/sling-org-apache-sling-feature-cpconverter/pull/74/commits/54c321f0f178e6adef4f2e95f853e2bb0231a27c.